ci/quality-base.sh

```diff
@@ -1,55 +1,54 @@
 #!/bin/bash
 
-source pipx-base.sh
+source uvx-base.sh
 
 python_files_and_folders() {
-    echo $(python $(script_dir)/python_files_and_folders.py)
+    echo $(.venv/bin/python $(script_dir)/python_files_and_folders.py)
 }
 
 run_ruff() {
     PYTHON_FILES_AND_FOLDERS=$(python_files_and_folders)
-    run_pipx ruff check $PYTHON_FILES_AND_FOLDERS
-    run_pipx ruff format --check $PYTHON_FILES_AND_FOLDERS
+    run_uvx ruff check $PYTHON_FILES_AND_FOLDERS
+    run_uvx ruff format --check $PYTHON_FILES_AND_FOLDERS
 }
 
 run_fixit() {
-    run_pipx fixit lint $(python_files_and_folders)
+    run_uvx fixit lint $(python_files_and_folders)
 }
 
 run_mypy() {
     PYTHON_FILES_AND_FOLDERS=$(python_files_and_folders)
     # Run mypy with or without pydantic plugin depending on whether pydantic is listed as dependency in the tools
     # section of the optional dependencies in the pyproject.toml file.
-    pydantic_spec=$(spec pydantic)
+    pydantic_spec=$(spec pydantic ==)
     if [[ "$pydantic_spec" == "" ]]; then
-        run_pipx mypy --python-executable=$(which python) $PYTHON_FILES_AND_FOLDERS
+        run_uvx mypy --python-executable=.venv/bin/python $PYTHON_FILES_AND_FOLDERS
     else
-        # To use the pydantic plugin, we need to first install mypy and then inject pydantic
-        run pipx install --force $(spec mypy) # --force works around this bug: https://github.com/pypa/pipx/issues/795
-        run pipx inject mypy $pydantic_spec
-        run $PIPX_BIN_DIR/mypy --python-executable=$(which python) $PYTHON_FILES_AND_FOLDERS
+        run uvx --with $pydantic_spec $(spec mypy @) --python-executable=.venv/bin/python $PYTHON_FILES_AND_FOLDERS
```

Comment:
I don't get it, doesn't this produce something like `--with pydantic==1.2.3 mypy@4.5.6` ?

Reply:
Yup!